internal/services/automation/automation_account_resource.go

```diff
@@ -193,17 +198,19 @@ func resourceAutomationAccountRead(d *pluginsdk.ResourceData, meta interface{})
 		return fmt.Errorf("retrieving Registration Info for %s: %+v", *id, err)
 	}
 
-	d.Set("name", id.Name)
-	d.Set("resource_group_name", id.ResourceGroup)
-	d.Set("location", location.NormalizeNilable(resp.Location))
+	d.Set("name", id.AutomationAccountName)
+	d.Set("resource_group_name", id.ResourceGroupName)
+
+	d.Set("location", location.NormalizeNilable(resp.Model.Location))
 	publicNetworkAccessEnabled := true
-	if resp.PublicNetworkAccess != nil {
-		publicNetworkAccessEnabled = *resp.PublicNetworkAccess
+	prop := resp.Model.Properties
+	if prop.PublicNetworkAccess != nil {
+		publicNetworkAccessEnabled = *prop.PublicNetworkAccess
 	}
 	d.Set("public_network_access_enabled", publicNetworkAccessEnabled)
 	skuName := ""
-	if sku := resp.Sku; sku != nil {
-		skuName = string(resp.Sku.Name)
+	if sku := prop.Sku; sku != nil {
+		skuName = string(prop.Sku.Name)
 	}
 	d.Set("sku_name", skuName)
```

Comment:
it's possible for `resp.Model` to be nil, as such we're using the following:

```
if model := resp.Model; model != nil {
  if props := model.Properties; props != nil {
    // ..
  }
}
```
the absence of a model allows us to determine that there's a larger issue which needs to be investigated - as all of the Azure-populated fields will be missing at Import time